src/backend/distributed/connection/placement_connection.c

```diff
@@ -559,6 +559,8 @@ StartColocatedPlacementConnection(uint32 flags, ShardPlacement *placement,
 	 */
 	if (placement->partitionMethod != DISTRIBUTE_BY_HASH)
 	{
+		// TODO: This is where we place a function mapping groupid -> nodename, nodeport
+		// flags can later tell us whether secondaries are okay
```

Comment:
Can't the nodename/nodeport in ShardPlacement just reflect the name and port of the current primary of the group? Renaming it could make sense. Removing it seems impractical given how many code paths rely on those variables.